### PR TITLE
Supplemental Claim | Fix confirmation page initial focus for Safari

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > After merging in the confirmation page focus code in #24414, it was noted that the focus is Safari was not working. After some testing, it started working after adding a slight delay after the `va-alert` web component had rendered.
- _(If bug, how to reproduce)_
  > Submit a Supplemental Claim in Safari and check confirmation page initial focus
- _(What is the solution, why is this the solution)_
  > Safari seems to be a special case here... the focus was working in Chrome, Firefox and Brave when tested
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#59546](https://github.com/department-of-veterans-affairs/va.gov-team/issues/59546)

## Testing done

- _Describe what the old behavior was prior to the change_
  > Added delay before focusing on `H2` inside `va-alert`
- _Describe the steps required to verify your changes are working as expected_
  > Test Supplemental Claim submission in Safari
- _Describe the tests completed and the results_
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

After
Focus outline shown on header inside the va-alert

## What areas of the site does it impact?

Supplemental Claim

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/995/containers/ConfirmationPage.jsx [hunk 1]
```diff
@@ -12,111 +12,118 @@ import { FORMAT_READABLE } from '../constants';
...
+        {fullName ? (
+          <div>
+            {name.first} {name.middle} {name.last}
+            {name.suffix ? `, ${name.suffix}` : null}
+          </div>
+        ) : null}
+
+        {submitDate.isValid() ? (
+          <>
+            <p />
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/containers/ConfirmationPage.jsx:75]: Does this format make it so we don't need a reguar `<p> </p>` tag?

**Review by rocio-desantiago — COMMENTED** [review]
P tag question

File: src/applications/appeals/995/containers/ConfirmationPage.jsx [hunk 2]
```diff
@@ -12,111 +12,118 @@ import { FORMAT_READABLE } from '../constants';
...
-            </div>
-          ) : null}
+  useEffect(
+    () => {
+      if (alertRef?.current) {
+        scrollTo('topScrollElement');
+        setTimeout(() => {
+          // delay focus for Safari
+          focusElement('h2', {}, alertRef.current);
+        }, 250);
```
Comment (adamwhitlock1) [inline src/applications/appeals/995/containers/ConfirmationPage.jsx:28]: We had a similar issue with one of our pages on Profile and I used the `waitForRenderThenFocus` with a low interval (i think 50) to make sure that focus was correct on Safari

Do you think there is a benefit in one over the other?

**Review by adamwhitlock1 — APPROVED** [review]

File: src/applications/appeals/995/containers/ConfirmationPage.jsx (hunk 2 above)
Comment (Mottie) [reply src/applications/appeals/995/containers/ConfirmationPage.jsx:28]: I didn't think to use that function LOL.. I think `waitForRenderThenFocus` might be better

File: src/applications/appeals/995/containers/ConfirmationPage.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/995/containers/ConfirmationPage.jsx:75]: Yes! I noticed a console warning about the header being inside the `<p>` tag, so I moved it above to keep the spacing.

**Pull request merged**